src/Pages/ProductList/ProductList.js

```diff
@@ -1,7 +1,164 @@
-import React, { Component } from "react";
+import React, { useState, useEffect } from "react";
+import styled from "styled-components";
+import ProductCategory from "./Components/ProductCategory/ProductCategory";
+import ClassificationBox from "./Components/ClassificationBox/ClassificationBox";
+import { PRODUCTLIST_API } from "../../config";
+import { AiOutlineHome } from "react-icons/ai";
 
-export default class ProductList extends Component {
-  render() {
-    return <div></div>;
-  }
+export default function ProductList() {
+  const [categories, setCategories] = useState([]);
+  // const [newCategories, setNewCategories] = useState([]);
+  const [copyCategories, setCopyCategories] = useState([]);
+  const [state, setState] = useState({
+    seeAllCategories: "",
+    subcategoryCheckboxId1: "",
+    // subcategoryCheckboxId2: "",
+    // subcategoryCheckboxId3: "",
+  });
+  const [checkedNames, setCheckedNames] = useState([]);
+
+  useEffect(() => {
+    // fetch(PRODUCTLIST_API)
+    fetch("/data/productList.json")
+      .then((res) => res.json())
+      .then((res) => setCategories(res.results));
+  }, []);
+
+  // 1. [인풋 업데이트 (카테고리 이름 업뎃)] & 검사mtd(비동기)
+  const isCheckedCategoryName = (e) => {
+    const { name, value, checked } = e.target;
+    setState({ [name]: value });
+    // setState({ [name]: checked });
+    makeCheckedNames(value);
+  };
+
+  // 2. [체크한 카테고리 배열에 넣는 mtd] (중복제거 mtd 포함)
+  const makeCheckedNames = (value) => {
+    const isInclude = checkedNames.some((name) => name === value);
+    const isIncludeAllBtn = checkedNames.some((name) => name === "전체 보기");
+
+    if (isInclude) {
+      const deleteOverlappedName = checkedNames.filter(
+        (name) => name !== value
+      );
+      setCheckedNames(deleteOverlappedName);
+    } else {
+      if (value === "전체 보기") {
+        setCheckedNames(["전체 보기"]);
+      } else {
+        if (isIncludeAllBtn) {
+          setCheckedNames([value]);
+          setCopyCategories([]);
+        } else {
+          setCheckedNames((prev) => [...prev, value]);
+        }
+      }
+    }
+  };
+
+  // 3. 하위 컴포넌트에서 체크된 카테고리만 나오게 만든 mtd
+  useEffect(() => {
+    alignData();
+  }, [checkedNames]);
+
+  // matchedDatas-> filter사용
+  const alignData = () => {
+    const isInclude = checkedNames.some(
+      // = e.target.value
+      (name) => name === state.subcategoryCheckboxId1
+    );
+    // const isIncludeAllBtn = checkedNames.some((name) => name === "전체 보기");
+
+    if (checkedNames.length > 0) {
```

Comment:
boolean값을 이용해 아래와 같이 작성해주세요
```suggestion
    if (checkedNames.length) {
```